course5/homework09/homework09.cc

```diff
@@ -0,0 +1,86 @@
+/*
+
+ARS/20170405
+
+Homework1:
+
+Use google benchmark to compare the performance of these two functions:
+
+bool is_greater(const std::string& a, const std::string& b) { return a > b; }
+bool is_greater(const char* a, const char* b) { return std::strcmp(a, b) > 0; }
+
+
+in a context when they are both called with const char* parameters. Test different length of the input string, from 1 char to 1MB (or even longer, if you so fancy or if results are interesting).
+
+Be mindful on what you’re measuring!
+
+Output:
+anil@anil-VirtualBox:~/cc/week9/debug$ ./homework09 
+Run on (1 X 3698.93 MHz CPU )
+2017-04-05 23:15:45
+***WARNING*** Library was built as DEBUG. Timings may be affected.
+-------------------------------------------------------------------------
+Benchmark                                  Time           CPU Iterations
+-------------------------------------------------------------------------
+is_greater_stdstring_loop/1               12 ns         12 ns   54107312
+is_greater_stdstring_loop/8               12 ns         12 ns   56405826
+is_greater_stdstring_loop/64              12 ns         12 ns   97076944
+is_greater_stdstring_loop/512             19 ns         19 ns   36336319
+is_greater_stdstring_loop/4096            12 ns         12 ns   56822159
+is_greater_stdstring_loop/32768           12 ns         12 ns   56628651
+is_greater_stdstring_loop/262144          12 ns         12 ns   56824993
+is_greater_stdstring_loop/1048576         12 ns         12 ns   56135011
```

Comment:
This isn't correct. Creating std::string from const char* requires copying of the data, so I'd expect this bencnhmark to be at least O(N). Your's clearly O(1)